packages/next/server/render.tsx

```diff
@@ -1307,25 +1307,27 @@ export async function renderToHTML(
     documentHTML = ReactDOMServer.renderToStaticMarkup(document)
   }
 
-  const nonRenderedComponents = []
-  const expectedDocComponents = ['Main', 'Head', 'NextScript', 'Html']
+  if (process.env.NODE_ENV !== 'production') {
+    const nonRenderedComponents = []
+    const expectedDocComponents = ['Main', 'Head', 'NextScript', 'Html']
 
-  for (const comp of expectedDocComponents) {
-    if (!(docComponentsRendered as any)[comp]) {
-      nonRenderedComponents.push(comp)
+    for (const comp of expectedDocComponents) {
+      if (!(docComponentsRendered as any)[comp]) {
+        nonRenderedComponents.push(comp)
+      }
     }
-  }
 
-  if (nonRenderedComponents.length) {
-    const missingComponentList = nonRenderedComponents
-      .map((e) => `<${e} />`)
-      .join(', ')
-    const plural = nonRenderedComponents.length !== 1 ? 's' : ''
-    throw new Error(
-      `Your custom Document (pages/_document) did not render all the required subcomponent${plural}.\n` +
-        `Missing component${plural}: ${missingComponentList}\n` +
-        'Read how to fix here: https://nextjs.org/docs/messages/missing-document-component'
-    )
+    if (nonRenderedComponents.length) {
+      const missingComponentList = nonRenderedComponents
+        .map((e) => `<${e} />`)
+        .join(', ')
+      const plural = nonRenderedComponents.length !== 1 ? 's' : ''
+      throw new Error(
```

Comment:
We should use `warn` instead of throwing an `Error`. It would crash in development but would be fine in production. That might be confusing.

Reply:
@balazsorban44 I've already fixed what you mentioned, let me know if it's ok and I will proceed with the PR.

thanks in advance.